File: OS HW1/Given Tests/hw1_test.c
#include <stdio.h>
#include <assert.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <unistd.h>
#include <string.h>
#include <errno.h>
#include "magic_api.h"

int my_strnlen(const char* str, int max_len) {
    int len = 0;
    while (len < max_len && str[len] != '\0') {
        len++;
    }
    return len;
}

int main()
{
	int res = -1;
    pid_t pid = getpid();
    char secrets[5][SECRET_MAXSIZE];

    // Simple test that uses the system calls
	res = magic_get_wand(1, "secret");
	assert(res == 0);

    res = magic_attack(pid);
    assert(res == -1);
    assert(errno == ECONNREFUSED);

    res = magic_legilimens(pid);
    assert(res == 0);

    res = magic_list_secrets(secrets, 5);
    assert(res == 0);
    int length = strlen(secrets[0]);//
    printf("length is: %d\n", length);//
    assert(my_strnlen(secrets[0], SECRET_MAXSIZE) == 0);

	return 0;
}

File: OS HW1/Given Tests/hw1_test.py
#!/usr/bin/python

import os
import errno
import pyMagic

def test1():
	"""simple test to invoke the syscalls """
	
	pid = os.getpid()

	pyMagic.get_wand(1, "secret")
	
	# Try to attack ourselves
	try:
		pyMagic.attack(pid)
	except OSError, e:
		assert(e.errno == errno.ECONNREFUSED)

	pyMagic.legilimens(pid)
	secrets, remaining = pyMagic.list_secrets(5)
	assert remaining == 0
	assert len(secrets) == 0

if __name__ == "__main__":
	test1()
